tools/gn

```diff
@@ -342,6 +342,17 @@ def to_gn_args(args):
     gn_args['host_cpu'] = 'x86'
     gn_args['current_cpu'] = 'x86'
 
+  # When building binaries to run on a macOS host (like gen_snapshot), always
+  # use the x64 toolchain. This is done for two reasons:
+  #  1. goma currently only supports the clang_x64 toolchain.
+  #  2. gen_snapshot cannot crossbuild from arm64 to x64. Its host architecture
+  #     must be x64 to target x64.
+  # However, if native arm64 binaries are needed when not building the engine
+  # for macOS (see the logic below), pass the flag "--mac-cpu arm64".
+  # https://github.com/flutter/flutter/issues/103386 is also relevant here.
+  if get_host_os() == 'mac' and not args.mac_cpu == 'arm64':
+    gn_args['host_cpu'] = 'x64'
```

Comment:
Haven't I already covered this for host builds in lines 356-360?

Reply:
I simplified the logic, and with the simpler logic, hopefully the comment makes more sense now. PTAL.

Reply:
Looks good. Thanks!